### PR TITLE
Fix issue with using Publish-Module together with latest ModuleBuilder

### PR DESCRIPTION
**Pull request by johlju**
# Pull Request

### Changed

- Update the repository to always use the latest version of the module
  `ModuleBuilder`.

### Fixed

- Now the prerelease is cleaned so that it does not contain any dashes by
  removing any suffix after a dash, for example `pr054-0012' will be changed
  to just `pr054' as the prerelease string. This is due to a bug in the
  cmdlet `Publish-Module` together with a newer version of the module
  `ModuleBuilder`.

## Task list

- [x] The PR represents a single logical change. i.e. Cosmetic updates should go in different PRs.
- [x] Added an entry under the Unreleased section of in the CHANGELOG.md as per [format](https://keepachangelog.com/en/1.0.0/).
- [x] Local clean build passes without issue or fail tests (`build.ps1 -ResolveDependency`).
- [ ] Resource documentation added/updated in README.md.
- [ ] Resource parameter descriptions added/updated in README.md, schema.mof
      and comment-based help.
- [ ] Comment-based help added/updated.
- [ ] Localization strings added/updated in all localization files as appropriate.
- [ ] Examples appropriately added/updated.
- [ ] Unit tests added/updated. See [DSC Resource Testing Guidelines](https://github.com/PowerShell/DscResources/blob/master/TestsGuidelines.md).
- [ ] Integration tests added/updated (where possible). See [DSC Resource Testing Guidelines](https://github.com/PowerShell/DscResources/blob/master/TestsGuidelines.md).
- [ ] New/changed code adheres to [DSC Resource Style Guidelines](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md) and [Best Practices](https://github.com/PowerShell/DscResources/blob/master/BestPractices.md).

---
This change is [Reviewable](https://reviewable.io/reviews/gaelcolas/sampler/151)

### CODE REVIEW

**Comment by gaelcolas** [comment]
Although it's non-blocking, I'll repeat that the "common module" should actually be the content of Sampler module.
I know, that's a bigger change, but I want to make sure we are on the same page.

File: .build/tasks/Common.Functions.psm1 [hunk 1]
```diff
@@ -105,7 +111,20 @@ function Get-ModuleVersion
...
-        $moduleVersion = ($moduleVersion -split '\+', 2)[0]
+        $moduleVersionWithoutMetadata = ($ModuleVersion -split '\+', 2)[0]
+
+        <#
+            Parses the prerelease string to remove any suffix that is not
+            handled by the cmdlet Publish-Module.
+        #>
+        $moduleVersion, $preReleaseString = $moduleVersionWithoutMetadata -split '-', 2
+
+        if ($preReleaseString)
```
Comment (gaelcolas) [inline .build/tasks/Common.Functions.psm1:122]: non-blocking: I would have added a comment saying the error is from Publish-Module which does not support semver compliant pre-release strings.

**Comment by johlju** [comment]
Yes, we are on the same page. I thought of that when I made the change that we should move these helper functions into Public/Private folders. It just bigger effort then I had time with right now. But still in my back head, I actually looked in the Private folder and thought “oh right, this is used by the Plaster template currently”. I can create an issue so we can track it. We should definitely change this. Remind me when I have 6 weeks of vacation in 2 months and I will fix it.  😃 Good exercise to learn Plaster better. 

I will update the comment as you suggested.

**Comment by gaelcolas** [comment]
Hahaha. Yeah the two dummy functions should be removed from there and moved to template assets. No Biggie, will get to it at a later date.

**Pull request merged**